### PR TITLE
[py-spacy] added version 2.3.7

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-spacy/package.py [hunk 1]
```diff
@@ -21,9 +22,11 @@ class PySpacy(PythonPackage):
     depends_on('py-cymem@2.0.2:2.0.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spacy/package.py]: Wheel shouldn't be needed right?

File: var/spack/repos/builtin/packages/py-spacy/package.py [hunk 2]
```diff
@@ -21,9 +22,11 @@ class PySpacy(PythonPackage):
...
     depends_on('py-preshed@3.0.2:3.0.999', type=('build', 'run'))
     depends_on('py-murmurhash@0.28:1.0', type=('build', 'run'))
-    depends_on('py-thinc@7.4.0', type=('build', 'run'), when='@2.2.4:2.2.999')
-    depends_on('py-thinc@7.4.1', type=('build', 'run'), when='@2.3.0:')
+    depends_on('py-thinc@7.4.0',         type=('build', 'run'), when='@2.2.4:2.2.999')
+    depends_on('py-thinc@7.4.1',         type=('build', 'run'), when='@2.3.0:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spacy/package.py]:
```suggestion
    depends_on('py-thinc@7.4.1',         type=('build', 'run'), when='@2.3.0:2.3.6')
```

File: var/spack/repos/builtin/packages/py-spacy/package.py [hunk 3]
```diff
@@ -21,9 +22,11 @@ class PySpacy(PythonPackage):
...
+    depends_on('py-thinc@7.4.1:7.4.999', type=('build', 'run'), when='@2.3.7:')
     depends_on('py-blis@0.4.0:0.4.999', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-spacy/package.py]:
```suggestion
    depends_on('py-blis@0.4.0:0.4.999', type=('build', 'run'), when='@:2.3.6')
```

**Comment by bernhardkaindl** [comment]
@qwertos Are you still there? Can you apply the suggestions from Adam?

File: var/spack/repos/builtin/packages/py-spacy/package.py (hunk 1 above)
Comment (bernhardkaindl) [reply var/spack/repos/builtin/packages/py-spacy/package.py]: Rebased to develop and removed: Yep, not needed.

**Review by bernhardkaindl — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**